src/lib.rs

```diff
@@ -1 +1,29 @@
-pub struct SrTree {}
+use num_traits::Float;
+use num_traits::FromPrimitive;
+use num_traits::Zero;
+use std::ops::AddAssign;
+use std::ops::DivAssign;
+use std::ops::MulAssign;
+use std::ops::SubAssign;
+
+mod distance;
+mod rect;
+mod sphere;
+mod node;
+use crate::node::{Data, Node};
+
+pub struct SRTree<T, const dimension: usize> {
+    root: Option<node::Node<T, dimension>>,
+    m: usize, // min number of entries in a node / leaf
+    M: usize  // max number of entries in a node / leaf
```

Comment:
Follow the [Rust API Guidelines](https://rust-lang.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case) in naming. A variable name should be in small letters.